### PR TITLE
Dockerfile 

### LINKED ISSUE #1075: Talawa-Admin Docker Support Issue
### Talawa-Admin Docker Support Issue:

**Title:**
Adding Docker Support to Talawa-Admin

**Description:**

**Problem:**
Talawa-Admin currently lacks Docker support, hindering its ease of deployment and scalability. Adding Docker support is essential to streamline the deployment process and make it more accessible to a broader audience.

**Solution:**
Implementing Docker support for Talawa-Admin would involve creating a Dockerfile and configuration to package the application into a Docker container. This containerization would simplify deployment, improve consistency across environments, and enhance scalability.

**Approach:**
To achieve Docker support for Talawa-Admin, we can follow these steps:
1. Create a Dockerfile specifying the necessary dependencies and configuration.
2. Configure the application to work seamlessly within a Docker container.
3. Document the steps for building and running Talawa-Admin using Docker in the project's README.

**Additional Context:**
Adding Docker support aligns with modern development practices, making it easier for developers to set up and run Talawa-Admin in various environments. Docker containers provide isolation, portability, and reproducibility, contributing to a more robust and user-friendly development and deployment experience.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by VanshikaSabharwal**
**What kind of change does this PR introduce?**

 feature

**Issue Number:**

Fixes #1075

**Did you add tests for your changes?**
No

**Snapshots/Videos:**

image
image
image
image

**If relevant, did you update the documentation?**

**Summary**

-->Added a Dockerfile and a docker-compose.yaml file to containerize the application.
-->Modified config.vite.ts:
1. Set host: '0.0.0.0' to allow the application to be accessible from outside the Docker container.
2. Set open: false to prevent the app from automatically opening in the browser on startup.

These changes were necessary because the app wasn't initially accessible when run inside a Docker container. With these adjustments, talawa-admin can now be successfully accessed while running in Docker.

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced a `.dockerignore` file to optimize Docker image builds.
	- Added a new `Docker-Start-Check` job in the GitHub Actions workflow to verify the application starts correctly in Docker.
	- New `app` service in `docker-compose.yaml` for building the application.

- **Changes**
	- Updated server settings in `vite.config.ts` to modify accessibility and startup behavior.
	- Implemented a multi-stage build process in the Dockerfile for improved application deployment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN npm install -g typescript
+
+RUN npm install
+
+RUN npm run build
+
+EXPOSE 4321
+
+CMD ["npm", "run", "serve"]
```
[the remark this answers is not in the document]
Comment (VanshikaSabharwal) [reply Dockerfile:15]: did it. 
image

**Comment by palisadoes** [comment]
Please make code rabbit approve your changes.

You should close this PR and add the docker file changes to this PR 
- https://github.com/PalisadoesFoundation/talawa-admin/pull/2414

**Comment by palisadoes** [comment]
During the week of November 11, 2024 we will start a code freeze on the `develop` branches in Talawa, Talawa Admin and Talawa-API. 

We have completed a project to convert the Talawa-API backend to use PostgreSQL. Work will then begin with us merging code in the `develop` branches to a new `develop-postrgres` branch in each repository.

Planning activities for this will be managed in our #talawa-projects slack channel. A GitHub project will be created to track specially labeled issues. We completed a similar exercise last year using a similar methodology. 

Starting November 12, California time no new PRs will be accepted against the `develop` branch. They must be applied to the `develop-postrgres` branch. 

There are some GSoC project features that will need to be merged into develop. These will be the only exceptions.

This activity and the post GSoC 2024 start date was announced in our #general Slack channel last month as a pinned post.

**Comment by VanshikaSabharwal** [comment]
> Please make code rabbit approve your changes.
> 
> You should close this PR and add the docker file changes to this PR
> 
> * [added docker check to workflow #2414](https://github.com/PalisadoesFoundation/talawa-admin/pull/2414)

yes I did it.

**Pull request closed without merging**